src/components/events/Goals/index.js

```diff
@@ -0,0 +1,64 @@
+var React = require('react');
+var totals = require('../../../api/totals');
+var CampaignGoalItem = require('../GoalItem');
+
+module.exports = React.createClass({
+  displayName: 'CampaignGoals',
+
+  getDefaultProps: function() {
+    return {
+      campaigns: []
+    }
+  },
+
+  getInitialState: function() {
+    return {
+      campaigns: []
+    }
+  },
+
+  componentWillMount: function() {
+    var campaignTotals = [];
+    var campaignsLength = this.props.campaigns.length;
+    var handler = this.handleCampaignTotals;
+    this.props.campaigns.map(function(campaign) {
+      return totals.findByCampaigns({
+        campaignUids: campaign.uid
+      }, function(response) {
+        campaignTotals.push({
+          name: campaign.name,
+          goal: campaign.goal,
+          count: response.total_amount_cents.count
+        })
+
+        if (campaignTotals.length === campaignsLength) {
+          handler(campaignTotals);
+        }
+      })
+    })
```

Comment:
I feel like the logic in this method could be moved in to the API section, so that it may be reused